modules/cas_cache/cas_cache.h

```diff
@@ -96,4 +97,63 @@ struct casdsk_functions_mapper {
 	struct gendisk *(*casdsk_exp_obj_get_gendisk)(struct casdsk_disk *dsk);
 };
 
+static inline int cache_name_from_id(char *name, uint16_t id)
+{
+	int result;
+
+	result = snprintf(name, OCF_CACHE_NAME_SIZE, "cache%d", id);
+
+	return result < 0 ? result : 0;
+}
+
+static inline int core_name_from_id(char *name, uint16_t id)
+{
+	int result;
+
+	result = snprintf(name, OCF_CORE_NAME_SIZE, "core%d", id);
+
+	return result < 0 ? result : 0;
+}
+
+static inline int cache_id_from_name(uint16_t *cache_id, const char *name)
+{
+	const char *id_str = name + sizeof("cache")  - 1;
+	long res;
+	int result;
+
+	result = kstrtol(id_str, 10, &res);
+
+	if (!result)
+		*cache_id = res;
+
+	return result;
+}
+
+static inline int mngt_get_cache_by_id(ocf_ctx_t ctx, uint16_t id,
+		ocf_cache_t *cache)
+{
+	static char cache_name[OCF_CACHE_NAME_SIZE];
```

Comment:
Why static? This is susceptible to races when two or more threads call this function. This applies to get_core_by_id as well.

Or you can add the mutex instead of chaning this (https://trello.com/c/aMVuXF1n)

Reply:
This left from early implementation. Done.